### ISSUE TITLE
VSI-Bench_32frame results are not reproducible

### ISSUE THREAD

**Issue by zzzyf01** [EvolvingLMMs-Lab/EASI#20]
### Summary
I’m trying to reproduce the reported **VSI-Bench_32frame** results for the following two models, but the numbers I obtain do not match what’s shown/claimed (e.g., on the model card), and I can’t reproduce the published comparison.

Models:
- `sensenova/SenseNova-SI-1.1-Qwen3-VL-8B`  
  Model card: https://huggingface.co/sensenova/SenseNova-SI-1.1-Qwen3-VL-8B
- `cambrian-s-7b`
  Model card: https://huggingface.co/nyu-visionx/Cambrian-S-7B

### My reproduced results (VSI-Bench_32frame)
Using the same evaluation pipeline, the **overall accuracy** I get is:

- **SenseNova-SI-1.1-Qwen3-VL-8B: 55.73**
- **cambrian-s-7b: 62.95**

These are significantly different from the numbers implied by the data on the model card / reported comparison, so the results appear **non-reproducible** as-is.

### Sanity check: ViewSpatialBench
For **ViewSpatialBench**, results fluctuate within a **reasonable range** across runs. The main reproducibility issue seems specific to **VSI-Bench_32frame** (overall accuracy).

### Change I made to the evaluation script (avoid ChatGPT-as-a-judge)
I noticed the evaluation code may call `chatgpt-0125` as an external judge for some datasets. To avoid introducing non-determinism and dependency on an external service/model version, I added a condition in `run.py` around line **372** to prevent using `chatgpt-0125` and instead rely on the `extract_matching`-based evaluation path.

Patch:

```python
                        elif listinstr(['VSI-Bench', 'SparBench', 'ViewSpatialBench', 'CV-Bench', 'CVBench', 'SPBench'], dataset_name):
                            pass
```

**Comment by PeterWangyi** [comment]
@zzzyf01 Hi there, thanks for the interest in EASI the very detailed reproduction notes.

There are two main reasons behind the discrepancy you observe on `VSI-Bench_32frame`:

**1. Input modality (video vs. image).** 

SenseNova-SI-1.1-Qwen3-VL-8B is fine-tuned on top of Qwen3-VL using **image-only** data. For VSI-Bench (a video benchmark), our reported number is obtained by treating each video as **sampled frames fed through the image pathway** (multi-image input), not the video pathway.

In our setup:

--model `Qwen3-VL-8B` --data `VSI-Bench_32frame`
- video input: 55.59
- image input (sampled frames): 57.90

--model `SenseNova-SI-1.1-Qwen3-VL-8B` --data `VSI-Bench_32frame`
- video input: 55.20
- image input: 64.83

Your reproduced ~55.7 for SenseNova matches our video-input result (55.20).
The reported 64.83 is from the image-input setting, which is consistent with the model’s training modality.
To reproduce the image-input setting in VLMEval/EASI, you can switch Qwen3-VL to image mode, e.g.:
```
# in vlmeval/vlm/qwen3_vl/model.py (around L48)
video_llm = False
```

 **2. Judge mode (extract_matching vs. LLM judge).**
For the SenseNova-SI VSI-Bench results we report, we use judge_mode=extract_matching only.
The LLM-as-judge path is optional and not used for the published numbers.

In this setting we also checked that GPT-4o as judge gives the **same scores**:

SenseNova-SI-1.1-Qwen3-VL-8B, image input: extract_matching → 64.83, GPT-4o → 64.83
SenseNova-SI-1.1-Qwen3-VL-8B, video input: extract_matching → 55.20, GPT-4o → 55.20

So the gap comes from video vs image evaluation, not from the judge.
You can explicitly set the judge mode via the CLI, e.g.:
```
--judge extract_matching
# or
--judge your_api_model
```

**3. For completeness**
Cambrian-S-7B is reported in the paper **as 67.5 on VSI with 128 frames**.
On EASI we standardize to 32 frames, and our 32-frame results are close to the official repo’s 32-frame results:

Cambrian-S-3B: official 54.73 vs. EASI 56.08
Cambrian-S-7B: official 63.61 vs. EASI 62.93

All differences are within ~2 points, which we consider reasonable given environment / judge mode / inference params differences.

We’ll update the docs to clearly state, for each benchmark, (1) which input modality (video vs. image) is used and (2) which judge mode is used, so that results are easier to reproduce.

Again, thanks a lot for running these careful checks and for raising the reproducibility question, this is very valuable feedback

**Comment by zzzyf01** [comment]
Thank you very much for your response! This has resolved my confusion.
